### PR TITLE
Load ISC fields using ajax

### LINKED ISSUE #117: Prevent loading all image meta data
In https://github.com/image-source-control/image-source-control/blob/336e5a20ab77af958253dae0941966d638a52ba3/includes/gutenberg/gutenberg.php#L36, ISC loads the meta data for all images into a JS array. This could cause problems with many images.

I cannot remember why we did this, but my feeling would be that we did it to have the data available dynamically when an image source is added. Still, we might want to change that into using an AJAX request per image.

### PR DESCRIPTION
**Pull request by cnhk**
fix #117

### CODE REVIEW

**Comment by cnhk** [comment]
- One batch for all images at page load.
- Each inserted image use one extra ajax call.

**Review by Galanos — COMMENTED** [review]
Tested with 100 images in media library, looks good! Feels just as fast as before. Checked the following:
- [x] Media > Library overview
- [x] Media > Image Sources
- [x] Media > Library > (select random image) > (updated ISC fields)
- [x] Posts > All Posts > (edit existing post) > (insert image) > (select from Media Library) > (filled in ISC meta data)
- [x] Settings > Image Sources
- [x] Media > Image Sources > Debug > list post-image relations > (select random image) > Edit Media > (filled in ISC meta data)
- [x] (Global list on dedicated page in frontend)

File: includes/gutenberg/isc-image-block.js [hunk 1]
```diff
@@ -23,14 +23,126 @@
...
+		batchWaitTime      = 1000,
+		batchCallbacks = {};
+
+	/**
+	 * Queue an image ID and "setAttribute" function, to be called in batch in one single AJAX call.
+	 *
+	 * @param id
+	 * @param setAttributes
+	 */
+	var bacthLoad = function ( id, setAttributes ) {
```
Comment (webzunft) [inline includes/gutenberg/isc-image-block.js:53]: typo in `bacthLoad`

File: includes/gutenberg/isc-image-block.js [hunk 2]
```diff
@@ -23,14 +23,126 @@
...
+	 * @param setAttributes callback function to call when the AJAX call is completed.
+	 */
+	var loadImageMeta = function ( id, setAttributes ) {
+		if ( typeof currentMetaLoading[id] !== 'undefined' ) {
+			return;
+		}
+
+		if ( useBatch ) {
+			// The batch call is still cooking. Queue the current image.
+			bacthLoad( id, setAttributes );
```
Comment (webzunft) [inline includes/gutenberg/isc-image-block.js:107]: typo

**Review by webzunft — CHANGES_REQUESTED** [review]
Looks good. I just would like to get rid of that typo.

**Comment by webzunft** [comment]
If someone disables the REST API, wouldn’t they break the block editor completely? If they do, then the ISC block options don’t make sense for them, anyway. Or am I missing something?

**Comment by cnhk** [comment]
That's right, Gutenberg can't work without the REST API.

**Comment by webzunft** [comment]
@cnhk code looks good.
@Galanos would you mind testing the pull request on the test site, again?

**Review by webzunft — APPROVED** [review]

**Review by Galanos — COMMENTED** [review]
@webzunft Don’t mind at all! Again, tested with 100 images in media library. Checked the following again:
- [x] Media > Library overview
- [x] Media > Image Sources
- [x] Media > Library > (select random image) > (updated ISC fields)
- [x] Posts > All Posts > (edit existing post) > (insert image) > (select from Media Library) > (filled in ISC meta data)
- [x] Settings > Image Sources
- [x] Media > Image Sources > Debug > list post-image relations > (select random image) > Edit Media > (filled in ISC meta data)
- [x] (Global list on dedicated page in frontend)
Everything looking great. Didn’t notice any delays or other sluggish behavior.

**Pull request merged**